frontend/src/store/facilityBooking/action.ts

```diff
@@ -394,13 +401,115 @@ export const fetchSelectedFacility = (bookingId: number) => async (dispatch: Dis
   // await fetch(DOMAIN_URL.EVENT + ENDPOINTS.CCA_DETAILS + '/' + booking.ccaID, { method: 'GET', mode: 'cors' })
 }
 
+//TODO: Remove when edit booking is done
 export const setBookingRepeat = (numRepeatWeekly: number) => (dispatch: Dispatch<ActionTypes>) => {
   dispatch({
     type: FACILITY_ACTIONS.SET_REPEAT_WEEKLY,
     numRepeatWeekly: numRepeatWeekly,
   })
 }
 
+export const setBooking = (booking: Booking) => (dispatch: Dispatch<ActionTypes>) => {
+  dispatch({
+    type: FACILITY_ACTIONS.SET_BOOKING,
+    booking: booking,
+  })
+}
+
+export const resetBooking = () => (dispatch: Dispatch<ActionTypes>) => {
+  dispatch({
+    type: FACILITY_ACTIONS.SET_BOOKING,
+    booking: null,
+  })
+}
+
+export const handleCreateNewBooking = (
+  facilityID: number,
+  eventName: string,
+  startTime: number | null,
+  endTime: number | null,
+  endDate: number | null,
+  ccaID?: number,
+  description?: string,
+) => async (dispatch: Dispatch<ActionTypes>) => {
+  const requestBody = {
+    facilityID: facilityID,
+    eventName: eventName,
+    userID: localStorage.getItem('userID'),
+    ccaID: ccaID,
+    startTime: startTime,
+    endTime: endTime,
+    endDate: endDate,
+    description: description,
+    // TODO: add recurrence end date
```

Comment:
documentation should be up for this? do check; or is this for when recurring end date state is ready?